docs/configuration/configuration.md

```diff
@@ -918,6 +922,77 @@ instance it is running on should have at least read-only permissions to the
 compute resources. If running outside of GCE make sure to create an appropriate
 service account and place the credential file in one of the expected locations.
 
+### `<hetzner_sd_config>`
+
+Hetzner SD configurations allow retrieving scrape targets from [Hetzner Cloud](https://www.hetzner.cloud/)
+Server API.
+This service discovery uses the public IPv4 address by default, by that can be
+changed with relabelling, as demonstrated in [the Prometheus hetzner-sd
+configuration file](/documentation/examples/prometheus-hetzner.yml).
+
+The following meta labels are available on all targets during [relabeling](#relabel_config):
+
+* `__meta_hetzner_server_id`: the id of the server
+* `__meta_hetzner_server_name`: the name of the server
+* `__meta_hetzner_ipv4`: the public ipv4 of the server
+* `__meta_hetzner_datacenter`: the name of the datacenter the server is located
+
+
+The labels below are only available for targets with `platform` set to `cloud`
+* `__meta_hetzner_hcloud_image_name`: the image name or description of the server
+* `__meta_hetzner_hcloud_image_description`: the image name or description of the server
+* `__meta_hetzner_hcloud_location`: the name of the location the server is located
+* `__meta_hetzner_hcloud_type`: the name of the type of the server
+* `__meta_hetzner_hcloud_cpu_cores`: count of CPU cores the server has
+* `__meta_hetzner_hcloud_cpu_type`: the type of the CPU (shared or dedicated)
+* `__meta_hetzner_hcloud_memory_size`: amount of memory the server has (in GB)
+* `__meta_hetzner_hcloud_disk_size`: size of disk the server has (in GB)
+
+Additionally, if you configure the service discovery to look for a specific network
+the following meta labels are available:
+* `__meta_hetzner_hcloud_network`: the name of the network the server is attached to
+* `__meta_hetzner_hcloud_private_ipv4`: the private ipv4 of the server within this specific network
```

Comment:
The EC2 datasource is using meta api to take the current network id. Should we do this here too @brian-brazil ?

Reply:
Yeah, this sort of thing should be completely automatic.

Reply:
This will only work if the server where Prometheus is running will be in the same (private) network as the servers, otherwise our metadata service won’t recognize the network. A server within our cloud can be attached to 3 different networks, so you need to specify the network that you want to select.

Reply:
thanks for the info

Reply:
That sounds like business logic, so we should expose metadata rather than adding config options.